workers/unity/Packages/com.improbable.gdk.core/Tests/Editmode/GameObjectRepresentation/Readers/ReaderUpdateTests.cs

```diff
@@ -0,0 +1,269 @@
+                "Exceptions that happen within component update callbacks should not propagate to callers.");
+
+            Assert.IsTrue(secondUpdateCalled);
+        }
+
+        [Test]
+        public void FieldUpdates_get_called_if_a_field_changes()
+        {
+            bool floatFieldUpdated = false;
+
+            float receivedFloatValue = 0;
+
+            Reader.FloatFieldUpdated += newValue =>
+            {
+                floatFieldUpdated = true;
+                receivedFloatValue = newValue;
+            };
+
+            QueueUpdatesToEntity(new SpatialOSBlittableComponent.Update
+            {
+                FloatField = new Option<float>(10.0f),
+            });
+
+            var internalReader = (IReaderInternal) Reader;
+
+            internalReader.OnComponentUpdate();
+
+            Assert.IsTrue(floatFieldUpdated,
+                "The update contains a float field but the callback for the float field was not called.");
+            Assert.AreEqual(10.0f, receivedFloatValue);
+        }
+
+        [Test]
+        public void FieldUpdates_do_not_get_called_if_another_field_changes()
+        {
+            bool floatFieldUpdated = false;
+            bool intFieldUpdated = false;
+
+            float receivedFloatValue = 0;
+            int receivedIntValue = 0;
+
+            Reader.FloatFieldUpdated += newValue =>
+            {
+                floatFieldUpdated = true;
+                receivedFloatValue = newValue;
+            };
+            Reader.IntFieldUpdated += newValue =>
+            {
+                intFieldUpdated = true;
+                receivedIntValue = newValue;
+            };
+
+            QueueUpdatesToEntity(new SpatialOSBlittableComponent.Update
+            {
+                FloatField = new Option<float>(10.0f),
+            });
+
+            var internalReader = (IReaderInternal) Reader;
+
+            internalReader.OnComponentUpdate();
+
+            Assert.IsTrue(floatFieldUpdated,
+                "The update contains a float field but the callback for the float field was not called.");
+            Assert.AreEqual(10.0f, receivedFloatValue);
+            Assert.IsFalse(intFieldUpdated,
+                "The update does not contain an int field but the callback for the int field was called. ");
+
+            floatFieldUpdated = false;
+
+            ClearUpdatesInEntity();
+
+            QueueUpdatesToEntity(new SpatialOSBlittableComponent.Update
+            {
+                IntField = new Option<int>(20),
+            });
+
+            internalReader.OnComponentUpdate();
+
+            Assert.IsFalse(floatFieldUpdated,
+                "The second update does not contain a float field but the callback was called.");
+            Assert.IsTrue(intFieldUpdated,
+                "THe second update contains an int field but the callback for the int field was not called.");
+            Assert.AreEqual(20, receivedIntValue);
```

Comment:
Is the second half of this test required? The first half effectively tests the premise afaict